### PR TITLE
Enable Row Selection on Plugin Table Component

### PR DESCRIPTION
**Pull request by duytnguyendtn**
### Description

Hello Jdaviz Team!

This PR is a small change to the Plugin Table component to expose the Vuetify row selection attributes for the Virtual Observatory plugin I'm working on. Because this is a frontend component, no new tests were added. A changelog entry will follow the publication of this PR, once a PR number is assigned.

image

It was great to sync my fork again :)
Screenshot 2024-05-06 150559

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-4468)

### CODE REVIEW

**Comment by pllim** [comment]
But how do you grab the selected rows for further operations? Is a test possible for this?

Thanks!

**Comment by duytnguyendtn** [comment]
You can see an example of its intended usage in my VO plugin here: https://github.com/duytnguyendtn/SkyviewJdaviz/blob/pyvo/pyvo_plugin/pyvo.py#L128-L141, but in short, you access it via the `selected_rows` attribute on the table via:

```
user_selection = my_plugin.table.selected_rows
```

I am not aware of a programmatic way of selecting the rows, other than setting `selected_rows` directly. In which case, the test would just be setting, then immediately reading the same traitlet, which didn't seem like a useful test to me 😅

File: CHANGES.rst [hunk 1]
```diff
@@ -82,6 +82,8 @@ Specviz2d
 API Changes
 -----------
 
+- Plugin Table components now support row selection. [#2856]
+
```
Comment (kecnry) [inline CHANGES.rst:86]: since this isn't actually used anywhere, it might not need a changelog entry?

**Review by kecnry — APPROVED** [review]
Looks simple enough to me - thanks!  I'm not really sure there is a way to test this until there is an actual use-case either.  If we're worried we may not have a use-case, we could let this sit unmerged until there is one and then add test coverage there 🤷‍♂️

**Comment by pllim** [comment]
> we could let this sit unmerged

My understanding is that Duy would use this downstream, no?

File: CHANGES.rst (hunk 1 above)
Comment (duytnguyendtn) [reply CHANGES.rst:86]: Entirely up to y'all, just need the instruction and I can remove it!

**Comment by kecnry** [comment]
I have no objection to merging myself - this is pretty isolated and would be pretty easy to strip back out if for some reason it goes unused (which seems unlikely).  @duytnguyendtn - can you please make a note to try to add test coverage for this capability in the eventual PR that makes use of it?

File: CHANGES.rst (hunk 1 above)
Comment (pllim) [reply CHANGES.rst:86]: Since people can use this downstream, let's keep it.

**Review by pllim — APPROVED** [review]
Kyle, feel free to merge when that extra comment you asked for is added. Thanks, all!

File: jdaviz/core/template_mixin.py [hunk 2]
```diff
@@ -4362,6 +4362,12 @@ class Table(PluginSubcomponent):
     headers_avail = List([]).tag(sync=True)   # list of strings
     items = List().tag(sync=True)  # list of dictionaries, pass single dict to add_row
 
+    # NOTE: These UI features are not covered in test coverage. Plugins making use of
+    # this feature should ensure test coverage for their respective tables.
```
Comment (duytnguyendtn) [inline jdaviz/core/template_mixin.py:4366]: Added a note regarding test coverage here. Thanks!

**Pull request merged**